cr-examples/onnx/src/main/java/oracle/code/onnx/OnnxRuntime.java

```diff
@@ -0,0 +1,535 @@
+package oracle.code.onnx;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.lang.foreign.*;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.VarHandle;
+import java.lang.reflect.UndeclaredThrowableException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import oracle.code.onnx.ir.OnnxOp;
+
+import static java.lang.foreign.ValueLayout.*;
+
+public final class OnnxRuntime {
+
+    private static final Path LIB_PATH;
+    static {
+        String arch = System.getProperty("os.arch", "generic").toLowerCase(Locale.ENGLISH).startsWith("aarch64") ? "aarch64" : "x64";
+        String os = System.getProperty("os.name", "generic").toLowerCase(Locale.ENGLISH);
+        String libResource;
+        if (os.contains("mac") || os.contains("darwin")) {
+          libResource = "/ai/onnxruntime/native/osx-" + arch + "/libonnxruntime.dylib";
+        } else if (os.contains("win")) {
+          libResource = "/ai/onnxruntime/native/win-" + arch + "/libonnxruntime.dll";
+        } else if (os.contains("nux")) {
+          libResource = "/ai/onnxruntime/native/linux-" + arch + "/libonnxruntime.so";
+        } else {
+          throw new IllegalStateException("Unsupported os:" + os);
+        }
+        try (var libStream = OnnxRuntime.class.getResourceAsStream(libResource)) {
+            var libFile = File.createTempFile("libonnxruntime", "");
+            LIB_PATH = libFile.toPath();
+            Files.copy(libStream, LIB_PATH, StandardCopyOption.REPLACE_EXISTING);
+            libFile.deleteOnExit();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static final Linker LINKER = Linker.nativeLinker();
+
+    private static final AddressLayout ADDR_WITH_ADDR = ADDRESS.withTargetLayout(ADDRESS);
+    private static final AddressLayout ADDR_WITH_STRING = ADDRESS.withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, JAVA_BYTE));
+
+    private static final VarHandle VH_ADDRESS = ADDRESS.varHandle();
+
+    private static Environment ENV;
+
+    public static Environment defaultEnvironment() {
+        if (ENV == null) ENV = new OnnxRuntime().createEnv();
+        return ENV;
+    }
+
+    private final Arena         arena;
+    private final SymbolLookup  library;
+    private final MemorySegment runtimeAddress, ret;
+    private final MethodHandle  allocatorGetInfo,
```

Comment:
Can these be `static` for better performance?

Reply:
No with the current API design. The memory segment of the instance of obtained by `OrtGetApiBase()->GetApi((uint32_t)apiVersion)`

Reply:
Can they be @Stable ?

Reply:
We can make things more optimal (if possible) later one, after we get things working.